test/test-robots.cpp

```diff
@@ -165,6 +165,13 @@ TEST(RobotsTest, SkipMalformedLine)
     EXPECT_TRUE(robot.allowed("/no/colon/in/this/line", "agent"));
 }
 
+TEST(RobotsTest, RobotsUrl)
```

Comment:
I would change this to `RobotsHttpUrl` and then add another test for an `https` example to show that it preserves the schema.